version: "3.9"
services:
  redis:
    image: redis:latest
    restart: on-failure
    ports:
      - "6379:6379"
    networks:
      - bot_network
  bot:
    restart: on-failure
    build: ./bot
    hostname: bot
    ports:
      - "81:81"
    volumes:
      - ./bot:/usr/src/app
    depends_on:
      - db
      - redis
    networks:
      - bot_network
  db:
    image: postgres:14-alpine
    restart: on-failure
    hostname: db
    ports:
      - "5432:5432"
    volumes:
      - ~/apps/postgres:/var/lib/postgresql/data
    environment:
      - POSTGRES_PASSWORD=secret
      - POSTGRES_USER=secret
      - POSTGRES_DB=local_bot_db
    networks:
      - bot_network
  nginx:
    restart: on-failure
    build: ./nginx
    ports:
      - '80:80'
    depends_on:
      - db
      - redis
      - bot
    networks:
      - bot_network
networks:
  bot_network:
